mRemoteV1/UI/Controls/QuickConnectToolStrip.cs

```diff
@@ -203,7 +203,7 @@ private void cmbQuickConnect_ProtocolChanged(object sender, QuickConnectComboBox
         private void btnQuickConnect_DropDownItemClicked(object sender, ToolStripItemClickedEventArgs e)
         {
             SetQuickConnectProtocol(e.ClickedItem.Text);
-            btnQuickConnect_ButtonClick(this, e);
+            _cmbQuickConnect.Focus();
```

Comment:
I suggest focusing the text box only if it's not filled out:

string.IsNullOrEmpty(e.ClickedItem.Text) ? _cmbQuickConnect.Focus() : btnQuickConnect_ButtonClick(this, e);

This would allow the proper workflow in either scenario...